wagtail/admin/views/generic/models.py

```diff
@@ -294,6 +301,12 @@ def search_queryset(self, queryset):
         return queryset.filter(**filters)
 
     def _get_title_column(self, field_name, column_class=TitleColumn, **kwargs):
+        if not issubclass(column_class, ButtonsColumnMixin):
+            column_class = type(
+                column_class.__name__,
+                (ButtonsColumnMixin, column_class),
+                {"get_buttons": self.get_list_buttons},
```

Comment:
Are you sure this does the right thing with `self`? This is giving out "how can this possibly work" vibes to me :-) Here `self.get_list_buttons` is bound to an instance (therefore it's a function that doesn't take `self` as its first argument) but it's going to become an attribute of the new class (which would have to be a function that takes `self` as its first argument).

For the similar trick in the base ViewSet I sidestepped this by adding a wrapper function with the expected signature:

https://github.com/wagtail/wagtail/blob/a50cc707a2d1bfd05c0df5e4784312a4a34279af/wagtail/admin/viewsets/base.py#L63-L64

Reply:
It does the right thing, due to how `self` is bound to instance methods, similar to how the `get_queryset(self, request)` from `SnippetViewSet` can be passed as the `queryset` attribute of the `IndexView`. Although yes, it is kind of funny and I think your approach makes more sense & is safer.

Basically the `self` of the method has been bound to the `IndexView` instance, and that will not change regardless where you stick the method to. It effectively kind of becomes like a static method. See the following:

```py
class A:
    def hello(self):
        print(f"hello from A: {self}")

class B:
    def hello(self):
        print(f"hello from B: {self}")

a = A()
C = type("C", (B,), {"hello": a.hello})
C().hello()
# hello from A: <__main__.A object at 0x1077a64a0>
C.hello()
# hello from A: <__main__.A object at 0x1077a64a0>
```

Will change this to follow your approach instead.

Reply:
Done in 34284fe61d922ec45a24eda5988bc125a97a1665.